### PR TITLE
Added necessary basic auth option to examples

### PR DESCRIPTION
**Pull request by c15yi**

### CODE REVIEW

**Comment by sberyozkin** [comment]
@c15yi Hi, as far as I recall, basic authentication is enabled by default if it is the only authentication mechanism around, are those specific examples do not work if you don't use anything else like `oidc` ? That said, rather than trying to save on not typing that property, it is worth being explicit about enabling it

File: docs/src/main/asciidoc/security-properties.adoc [hunk 1]
```diff
@@ -48,12 +48,14 @@ They are configured with properties starting with
...
 [source,properties]
 ----
+quarkus.http.auth.basic=true <1>
 quarkus.security.users.file.enabled=true
 quarkus.security.users.file.users=test-users.properties
 quarkus.security.users.file.roles=test-roles.properties
 quarkus.security.users.file.realm-name=MyRealm
 quarkus.security.users.file.plain-text=true
 ----
+<1> To enable basic auth you have to enable it with this option.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-properties.adoc:58]:
```suggestion
<1> Enable basic authuthentication.
```

File: docs/src/main/asciidoc/security-properties.adoc [hunk 2]
```diff
@@ -111,6 +114,7 @@ quarkus.security.users.embedded.roles.stuart=admin,user
 quarkus.security.users.embedded.roles.jdoe=NoRolesUser
 quarkus.security.users.embedded.roles.noadmin=user
 ----
+<1> To enable basic auth you have to enable it with this option.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-properties.adoc:117]:
```suggestion
<1> Enable basic authentication.
```

**Comment by sberyozkin** [comment]
@michalvavrik Hi Michal, I think it is OK to encourage users to enable basic authentication with the property even if it may not be necessary, as it can be a bit confusing otherwise, for example, it works now but then stops working the moment you added another mechanism, what do you think ?

**Comment by michalvavrik** [comment]
> @michalvavrik Hi Michal, I think it is OK to encourage users to enable basic authentication with the property even if it may not be necessary, as it can be a bit confusing otherwise, for example, it works now but then stops working the moment you added another mechanism, what do you think ?

I'd expect basic auth to be implicitly enabled, however it depends on user extensions / configuration. I don't know other mechanisms present. Problem I can see with changes is that this `security-properties-file` is about providing users/pwd/roles to Elytron identity provider. And they can be used by basic auth, form auth, custom auth accepting username/pwd creds, mTLS. Even token? I think we don't encourage that one.

TL;DR; I think it's fine as long as text makes it clear this is one of options and you have others. IMHO little tweak of added text is necessary.

**Comment by sberyozkin** [comment]
@michalvavrik 

>And they can be used by basic auth, form auth, custom auth accepting username/pwd creds, mTLS

Actually, that is a good point indeed. So I'm not sure what to do with this PR.

@c15yi Did you find you had to enable it because you were combining Basic with OIDC ?

May be we should have a Note added instead here, clarifying that if you use these properties to support Basic authentication mechanism, then you must explicitly enable basic authentication with this property if you also use other authentication mechanisms

**Comment by c15yi** [comment]
In our project we also use OIDC and basic auth was on top. So I guess, my change in the documentation is not necessary. I didn't know, that basic auth is automatically enabled, but only iff there is no other auth mechanism configured. 

I updated my change, to mention the default behaviour.

**Comment by sberyozkin** [comment]
Hi @c15yi Thanks, but I think it needs to be presented differently, add a decicated `Note`, you can find a lot of examples how to add Note, and only in that note clarify that `Basic authentication must be explicitly enabled with quarkus.http.basic=true if more than one authentication mechanism is used', there is no need to modify the actual properties, since as @michalvavrik has clarified, the same set of properties can be used for Form and other non Basic mechanisms where a user name and password are available

**Comment by gsmet** [comment]
@sberyozkin any chance you could drive this puppy home before tomorrow evening?

**Comment by sberyozkin** [comment]
Sorry @gsmet, noticed your ping only now that @c15yi pushed the update, thanks @c15yi :+1:

**Review by sberyozkin — APPROVED** [review]
